python/mxnet/lr_scheduler.py

```diff
@@ -74,3 +74,52 @@ def __call__(self, num_update):
             logging.info("Update[%d]: Change learning rate to %0.5e",
                          num_update, self.base_lr)
         return self.base_lr
+
+class MultiFactorScheduler(LRScheduler):
+    """Reduce learning rate in factor at steps specified in a list
+
+    Assume the weight has been updated by n times, then the learning rate will
+    be
+
+    base_lr * factor^(sum((step/n)<=1)) # step is an array
+
+    Parameters
+    ----------
+    step: list of int
+        schedule learning rate after n updates
+    factor: float
+        the factor for reducing the learning rate
+    """
+    def __init__(self, step, factor=1):
+        super(MultiFactorScheduler, self).__init__()
+        assert isinstance(step, list) and len(step) >= 1
+        for i, _step in enumerate(step):
+            if i != 0 and step[i] <= step[i-1]:
+                raise ValueError("Schedule step must be an increasing integer list")
+            if _step < 1:
+                raise ValueError("Schedule step must be greater or equal than 1 round")
+        if factor >= 1.0:
```

Comment:
Which one, _step < 1 or len(step)>= 1? It seems that there is no problem on my local computer.